### PR TITLE
GL4 Translator: ADDs/MULs/SUBs XZ instead of XY. Loops on FLOW_CONTROL.

### PR DESCRIPTION
**Pull request by DarioSamo**
Getting some good results with games that involve rigging with FLOW_CONTROL on.
Sonic Unleashed Demo: http://i.imgur.com/mndwqWd.jpg

### CODE REVIEW

**Comment by MoochMcGee** [comment]
Why is this behind a macro? If it's a non-hacky improvement, why isn't it just there in the code?

**Comment by DarioSamo** [comment]
The macro was already there behind flow control for stuff like conditional jumps, loops being implemented outside of the macro would have no effect on anything else.

The only one I have doubts about is the one that marks whether to add the relative offsets to the constants or not, so I left it behind the macro for now.

File: src/xenia/gpu/gl4/gl4_shader_translator.cc [hunk 1]
```diff
@@ -783,7 +800,7 @@ bool GL4ShaderTranslator::TranslateALU_ADDs(const instr_alu_t& alu) {
   AppendScalarOpSrcReg(alu, 3);
   Append(".x + ");
   AppendScalarOpSrcReg(alu, 3);
-  Append(".y");
+  Append(".z");
```
Comment (benvanik) [inline src/xenia/gpu/gl4/gl4_shader_translator.cc]: you sure z is right here?

**Comment by DarioSamo** [comment]
Most of the swizzles I've seen from the translator convert those types of instructions like r0.yw to r0.yyww, so x and z would be correct in this case. I do have my doubts about that one since I don't have many games to test with tho.

**Pull request merged**